[Win] Fix problems of SystemInfo.cpp

fujii opened the pull request:
#### d0795f0c2fa2986c03472c1ed35bf7d17c02fe5a
<pre>
[Win] Fix problems of SystemInfo.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=280295">https://bugs.webkit.org/show_bug.cgi?id=280295</a>

Reviewed by Ross Kirsling.

GetVersionEx API always returned the Windows 8 version number.
&lt;<a href="https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversionexa">https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversionexa</a>&gt;
Added supportedOS to application manifests to make the API work as
expected on newer Windows.

GetNativeSystemInfo API is available after Windows XP and Windows
Server 2003.
&lt;<a href="https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getnativesysteminfo">https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getnativesysteminfo</a>&gt;
We no longer need to use GetProcAddress for it.

Removed the code for old Windows in WebCore::createSharedCursor.

* Source/WebCore/platform/win/CursorWin.cpp:
(WebCore::createSharedCursor):
* Source/WebCore/platform/win/SystemInfo.cpp:
(WebCore::windowsVersion):
(WebCore::processorArchitecture):
(WebCore::architectureTokenForUAString):
(WebCore::windowsVersionForUAString):
(WebCore::osVersionForUAString): Deleted.
* Source/WebCore/platform/win/SystemInfo.h:
* Source/WebKit/win/WebKit.manifest:
* Tools/MiniBrowser/win/MiniBrowser.exe.manifest:
* Tools/WebKitTestRunner/win/WebKitTestRunner.exe.manifest:

Canonical link: <a href="https://commits.webkit.org/285009@main">https://commits.webkit.org/285009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7613efca66af51129e2f105e1a0b7d5929ff7da4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71039 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50451 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23812 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75143 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22244 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73155 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58250 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22062 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56168 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14644 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74105 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45831 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61228 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36611 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42478 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/18667 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/20585 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64415 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19026 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76863 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15271 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18215 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63907 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15315 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61291 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63864 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11971 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5609 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10919 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46252 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1028 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47324 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48607 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47066 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rkirsling reviewed (approved):
Very nice cleanup! :clap:

fujii committed:
[Win] Fix problems of SystemInfo.cpp
https://bugs.webkit.org/show_bug.cgi?id=280295

Reviewed by Ross Kirsling.

GetVersionEx API always returned the Windows 8 version number.
<https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversionexa>
Added supportedOS to application manifests to make the API work as
expected on newer Windows.

GetNativeSystemInfo API is available after Windows XP and Windows
Server 2003.
<https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getnativesysteminfo>
We no longer need to use GetProcAddress for it.

Removed the code for old Windows in WebCore::createSharedCursor.

* Source/WebCore/platform/win/CursorWin.cpp:
(WebCore::createSharedCursor):
* Source/WebCore/platform/win/SystemInfo.cpp:
(WebCore::windowsVersion):
(WebCore::processorArchitecture):
(WebCore::architectureTokenForUAString):
(WebCore::windowsVersionForUAString):
(WebCore::osVersionForUAString): Deleted.
* Source/WebCore/platform/win/SystemInfo.h:
* Source/WebKit/win/WebKit.manifest:
* Tools/MiniBrowser/win/MiniBrowser.exe.manifest:
* Tools/WebKitTestRunner/win/WebKitTestRunner.exe.manifest:

Canonical link: https://commits.webkit.org/285009@main
